R/demographics-table.R

```diff
@@ -0,0 +1,326 @@
+    cont_df <- left_join(cont_table, cont_table_all, by = c("name", "level"))
+    cat_df <- left_join(cat_table[["data"]], cat_table_all, by = c("name", "level"))
+    cols_extra <- left_join(
+      cat_table[["cols_extra"]],
+      cat_table_all0[["cols_extra"]],
+      by = "level"
+    )
+  }
+  if(summarize_all && !summarize_span) {
+    cont_df <- cont_table_all
+    cat_df <- cat_table_all
+    cols_extra <- cat_table_all[["cols_extra"]]
+  }
+  if(!summarize_all && summarize_span) {
+    cont_df <- cont_table
+    cat_df <- cat_table[["data"]]
+    cols_extra <- cat_table[["cols_extra"]]
+  }
+  # Combined Table ###
+  table_data <- bind_rows(cont_df, cat_df)
+  table_data <- rename(table_data, !!sym(stat_name) := .data[["level"]])
+  if(summarize_all) {
+    table_data <- rename(table_data, !!sym(all_name) := .data[["value"]])
+  }
+  # add units
+  units <- validate_units(units, data)
+  if(!is.null(units)) {
+    all_cols <- c(cols_cont, cols_cat)
+    has_unit <- match(names(units), all_cols)
+    nw <- names(all_cols)
+    nw[has_unit] <- paste(nw[has_unit],  unlist(units))
+    names(nw) <- names(all_cols)
+    table_data[["name"]] <- nw[table_data[["name"]]]
+  }
+  align <- cols_left()
+  ragged <- list(col_ragged(stat_width))
+  names(ragged) <- stat_name
+  align[["update"]] <- ragged
+  ans <- list(
+    span = new_span,
+    panel = as.panel("name"),
+    align = align,
+    cols_extra = cols_extra,
+    notes = notes
+  )
+  if(!paneled) {
+    ans[["panel"]] <- as.panel(NULL)
+    ans[["clear_reps"]] <- "Covariate"
+    if(!is.null(new_span)) {
+      ans[["cols_extra"]] <- bind_cols(
+        tibble(Covariate = ""),
+        ans[["cols_extra"]]
+      )
+    }
+    ans[["hline_from"]] <- "Covariate"
+    table_data <- select(
+      table_data,
+      Covariate = .data[["name"]],
+      everything()
+    )
+  }
+  ans[["data"]] <- table_data
+  structure(ans, class = c("pmtable", class(ans)))
+}
+
+demo_summarize_cont <- function(data, span, cols, fun) {
+  summary_names <- names(fun(1:5))
+  cont_table <- group_by(data, .data[["name"]], !!sym(span))
+  cont_table <- summarise(
+    cont_table,
+    fun(.data[["value"]]),
+    .groups = "drop"
+  )
+  cont_table <- mutate_at(cont_table, .vars = summary_names, as.character)
+  cont_table <- pivot_wider(
+    cont_table,
+    names_from  = "name",
+    values_from = summary_names,
+    names_glue  = "{name}_{.value}"
+  )
+  cont_table <- pivot_longer(cont_table, -!!sym(span))
+  cont_table <- pivot_wider(cont_table, names_from = all_of(unname(span)))
+  cont_table <- separate(
+    cont_table,
+    .data[["name"]],
+    c("name", "level"),
+    sep = "_"
+  )
+  cont_table <- mutate(
+    cont_table,
+    name = factor(.data[["name"]], levels = unname(cols))
+  )
+  cont_table <- arrange(cont_table, .data[["name"]])
+  cont_table <- mutate(cont_table, name = names(cols)[.data[["name"]]])
+  cont_table
+}
+
+pt_demographics_notes <- function() {
```

Comment:
Is `n: number of records summarized` included when it appears in the column header?

Reply:
Yes; I added a note there.